ux.symfony.com/src/Service/CookbookFactory.php

```diff
@@ -12,27 +12,65 @@
+use League\CommonMark\MarkdownConverter;
 
-class CookbookFactory
+final class CookbookFactory
 {
-    public function __construct(
-        private readonly CookbookParser $cookbookParser,
-        private readonly UrlGeneratorInterface $urlGenerator,
-    ) {
-    }
+    private MarkdownConverter $converter;
 
-    public function buildFromFile(\SplFileInfo $file): Cookbook
+    public function createFromFile(string $file): Cookbook
     {
-        $content = $file->getContents();
+        if (!file_exists($file)) {
+            throw new \InvalidArgumentException(\sprintf('File "%s" not found.', $file));
+        }
+
+        $content = file_get_contents($file);
+
+        $parsedContent = $this->getConverter()->convert($content);
+        if (!$parsedContent instanceof FrontMatterProviderInterface) {
+            throw new \RuntimeException('Unable to parse the cookbook.');
+        }
+
+        if (!\is_array($frontMatter = $parsedContent->getFrontMatter())) {
+            throw new \RuntimeException(\sprintf('Cookbook file "%s" does not contains Front Matter data.', $file));
+        }
+        if (!isset($frontMatter['title']) || !\is_string($frontMatter['title'])) {
+            throw new \RuntimeException('Missing title in Front Matter.');
+        }
+        if (!isset($frontMatter['description']) || !\is_string($frontMatter['description'])) {
+            throw new \RuntimeException('Missing description in Front Matter.');
+        }
+        if (!isset($frontMatter['image']) || !\is_string($frontMatter['image'])) {
+            throw new \RuntimeException('Missing image in Front Matter.');
+        }
+        if (!isset($frontMatter['tags']) || !\is_array($frontMatter['tags'])) {
+            throw new \RuntimeException('Missing tags in Front Matter.');
+        }
 
         return new Cookbook(
-            title: $this->cookbookParser->getTitle($content),
-            description: $this->cookbookParser->getDescriptions($content),
-            route: $this->urlGenerator->generate('app_cookbook_show', ['slug' => $file->getBasename('.md')]),
-            image: $this->cookbookParser->getImage($content),
-            content: $content,
-            tags: $this->cookbookParser->getTags($content),
+            title: $frontMatter['title'],
+            slug: str_replace('_', '-', basename($file, '.md')),
+            image: $frontMatter['image'],
+            description: $frontMatter['description'],
+            content: $parsedContent->getContent(),
+            tags: $frontMatter['tags'],
         );
     }
+
+    private function getConverter(): MarkdownConverter
+    {
+        if (isset($this->converter)) {
+            return $this->converter;
+        }
+
+        $environment = new Environment();
+        $environment->addExtension(new CommonMarkCoreExtension());
+        $environment->addExtension(new FrontMatterExtension());
+
+        return $this->converter = new MarkdownConverter($environment);
```

Comment:
Very minor, but I find this more easily understandable with this way:
```suggestion
        if (!isset($this->converter)) {
            $environment = new Environment();
            $environment->addExtension(new CommonMarkCoreExtension());
            $environment->addExtension(new FrontMatterExtension());

            $this->converter = new MarkdownConverter($environment)
        }

        return $this->converter;
```